src/v/cluster/partition.cc

```diff
@@ -537,7 +540,7 @@ partition::timequery(storage::timequery_config cfg) {
             local_query_cfg.min_offset = std::max(
               log()->from_log_offset(_raft->start_offset()),
               local_query_cfg.min_offset);
-            co_return co_await local_timequery(local_query_cfg);
+            co_return co_await local_timequery(local_query_cfg, false);
```

Comment:
nit: mind commenting that we're explicitly setting false because there's nothing to fall back on

Reply:
I'll do that in a separate PR. Need to review some additional things pointed out by Willem.